app/src/androidTest/java/ca/dal/cs/athletemonitor/athletemonitor/AccountManagerTest.java

```diff
@@ -0,0 +1,291 @@
+        DatabaseReference usersReference = database.getReference("online_users/" + testUser.getUsername());
+
+        //attach a listener for data changes of the users reference.  this will occur when
+        //the reference is populated
+        usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                //if the reference exists, then the user is tagged as logged in and we should fail
+                //the test, otherwise it should succeed
+                assertFalse("Expecting non-empty result from database, but no data returned...", dataSnapshot.exists());
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+    }
+
+    /**
+     * Tests creation of a new user
+     *
+     * Will pass if new account is added to the users branch of database
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void createUserTest() throws Exception {
+       final User testUser = TestingHelper.createTestUser();
+
+       AccountManager.createUser(testUser, new AccountManager.UserObjectListener() {
+           @Override
+           public void onUserPopulated(User user) {
+               assertNotNull("Expecting valid user object, but seen null...", user);
+
+               //if the user was created successfully, remove before exiting
+               AccountManager.deleteUser(testUser, TestingHelper.assertTrueBooleanResult());
+           }
+       });
+    }
+
+    /**
+     * Test for true positive on existing user
+     *
+     * Will pass in a username that does exist in the user accounts to the validate method and
+     * assert that the return value of validate is true
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void userExistsTest() throws Exception {
+        final User testUser = TestingHelper.createTestUser();
+        AccountManager.createUser(TestingHelper.createTestUser());
+
+        AccountManager.userExists(testUser.getUsername(), new AccountManager.UserExistsListener() {
+            @Override
+            public void onUserExists(boolean result) {
+                assertTrue("Expecting account " + testUser.getUsername() +
+                        " to be found, but it was not...", result);
+
+                //delete the test user before exiting test
+                AccountManager.deleteUser(testUser, null);
+            }
+        });
+    }
+
+    /**
+     * Test for false positive on existing user
+     *
+     * Will pass if a username that does not exist in the user accounts is passed to the userExists
+     * method.  Asserts that the return value of is false
+     *
+     * @throws Exception Generic exception
+     */
+    @Test
+    public void userDoesNotExistTest() throws Exception {
+        User testUser = new User("i don't exist", "testPassword");
```

Comment:
Maybe change the username to `"doesnotexist"` instead, since maybe in the future this could fail some validator (a username would not have spaces or an apostrophe)